### ISSUE TITLE
Under some conditions webcam API completely hangs

### ISSUE THREAD

**Issue by sarxos** [sarxos/webcam-capture#249]
When testing motion detector feature I found a problem when running below code. The effect is webcam panel repainting running smoothly for some time (e.g. 6 seconds), but later it completely freezes. What is worst, whole application completely freezes as well when user tries to close it (because of circular wait on shootdown hook). The problematic element is atomic processor instance which waits for task to be worked by webcam processor inbound/outbound queues. It waits because it is designed to process atomic tasks which operates on native layer and return, but due to buggy implementation in a connection between WebcamDefaultDevice and NextFrameTask it spawns new task from the and in the middle of other one being in progress. This causes deadlock.

``` java
import java.awt.BasicStroke;
import java.awt.Color;
import java.awt.Graphics2D;
import java.awt.Point;
import java.awt.image.BufferedImage;
import java.io.IOException;

import javax.swing.JFrame;

import com.github.sarxos.webcam.Webcam;
import com.github.sarxos.webcam.WebcamMotionDetector;
import com.github.sarxos.webcam.WebcamPanel;
import com.github.sarxos.webcam.WebcamResolution;


/**
 * Detect motion. This example demonstrates how to use build-in motion detector
 * and motion listener to fire motion events.
 * 
 * @author Bartosz Firyn (SarXos)
 */
public class DetectMotionExample3 extends JFrame implements WebcamPanel.Painter {

    private static final long serialVersionUID = 1L;

    private final Webcam webcam;
    private final WebcamPanel panel;
    private final WebcamMotionDetector detector;

    public DetectMotionExample3() {

        setTitle("Motion Detector Demo");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        webcam = Webcam.getDefault();
        webcam.setViewSize(WebcamResolution.VGA.getSize());
        webcam.open(true);

        panel = new WebcamPanel(webcam, false);
        panel.setPainter(this);
        panel.start();

        detector = new WebcamMotionDetector(webcam);
        detector.setInterval(500); // one check per 500 ms
        detector.setPixelThreshold(20);
        detector.start();

        add(panel);

        pack();
        setVisible(true);
    }

    public static void main(String[] args) throws IOException {
        new DetectMotionExample3();
    }

    @Override
    public void paintPanel(WebcamPanel panel, Graphics2D g2) {
        panel.getDefaultPainter().paintPanel(panel, g2);
    }

    @Override
    public void paintImage(WebcamPanel panel, BufferedImage image, Graphics2D g2) {

        double s = detector.getMotionArea();
        Point cog = detector.getMotionCog();

        Graphics2D g = image.createGraphics();
        g.setColor(Color.WHITE);
        g.drawString(String.format("Area: %.2f%%", s), 10, 20);

        if (detector.isMotion()) {
            g.setStroke(new BasicStroke(2));
            g.setColor(Color.RED);
            g.drawOval(cog.x - 5, cog.y - 5, 10, 10);
        } else {
            g.setColor(Color.GREEN);
            g.drawRect(cog.x - 5, cog.y - 5, 10, 10);
        }

        g.dispose();

        panel.getDefaultPainter().paintImage(panel, image, g2);
    }
}
```

The problematic stack:

``` plain
"atomic-processor-1" daemon prio=10 tid=0x00007f4028002000 nid=0x26cd waiting on condition [0x00007f403616b000]
   java.lang.Thread.State: WAITING (parking)
    at sun.misc.Unsafe.park(Native Method)
    - parking to wait for  <0x00000000c38f8820> (a java.util.concurrent.SynchronousQueue$TransferQueue)
    at java.util.concurrent.locks.LockSupport.park(LockSupport.java:156)
    at java.util.concurrent.SynchronousQueue$TransferQueue.awaitFulfill(SynchronousQueue.java:714)
    at java.util.concurrent.SynchronousQueue$TransferQueue.transfer(SynchronousQueue.java:645)
    at java.util.concurrent.SynchronousQueue.put(SynchronousQueue.java:810)
    at com.github.sarxos.webcam.WebcamProcessor$AtomicProcessor.process(WebcamProcessor.java:58)
    at com.github.sarxos.webcam.WebcamProcessor.process(WebcamProcessor.java:130)
    at com.github.sarxos.webcam.WebcamTask.process(WebcamTask.java:38)
    at com.github.sarxos.webcam.ds.buildin.WebcamDefaultDevice$NextFrameTask.nextFrame(WebcamDefaultDevice.java:60)
    at com.github.sarxos.webcam.ds.buildin.WebcamDefaultDevice.updateFrameBuffer(WebcamDefaultDevice.java:437)
    at com.github.sarxos.webcam.ds.buildin.WebcamDefaultDevice.getImageBytes(WebcamDefaultDevice.java:194)
    at com.github.sarxos.webcam.ds.buildin.WebcamDefaultDevice.getImage(WebcamDefaultDevice.java:263)
    at com.github.sarxos.webcam.ds.cgt.WebcamGetImageTask.handle(WebcamGetImageTask.java:43)
    at com.github.sarxos.webcam.WebcamProcessor$AtomicProcessor.run(WebcamProcessor.java:71)
    at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
    at java.lang.Thread.run(Thread.java:662)
```
